cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/SystemArguments.java

```diff
@@ -357,12 +357,23 @@ public static Resources getResources(Map<String, String> args, @Nullable Resourc
    * @return the profile id for the program run
    */
   public static ProfileId getProfileIdForProgram(ProgramId programId, Map<String, String> args) {
-    if (programId.getType() == ProgramType.WORKFLOW) {
+    if (isProgramTypeAllowedForProfile(programId.getType())) {
       return getProfileIdFromArgs(programId.getNamespaceId(), args).orElse(ProfileId.NATIVE);
     }
     return ProfileId.NATIVE;
   }
 
+  /**
+   * Returns true if the program type can use profiles, otherwise returns false.
```

Comment:
No need to repeat the doc with the `@return` tag. The javadoc for this method is probably better as `Determines if compute profile is allowed to be applied on the given program type.`, or something along that line.